app/handlers/sessions_callback.rb

```diff
@@ -39,11 +39,11 @@ def authorized?
   end
 
   def handle
-    # Get an authentication object for the incoming data, tracking if we have
+    # Get an authentication object for the incoming data, tracking if
     # the object didn't yet exist and we had to create it.
 
-    authentication_data = { provider: @auth_data[:provider],
-                            uid: @auth_data[:uid]}
+    authentication_data = { provider: @data.provider,
```

Comment:
actually don't need this `authentication_data` right?  Maybe we used to use it, but I don't see it below anywhere, so why not just pass these values into the `where` call?